bindings/go/oci/internal/lister/component/component_versions.go

```diff
@@ -76,21 +77,21 @@ func ReferenceTagVersionResolver(component string, store interface {
 		case ociImageSpecV1.MediaTypeImageManifest:
 			data, err = store.Fetch(ctx, desc)
 			if err != nil {
-				return "", fmt.Errorf("failed to fetch descriptor for tag %q: %w", tag, err)
+				return "", errors.Join(fmt.Errorf("failed to fetch descriptor for tag %q: %w", tag, err), data.Close())
```

Comment:
store and most ReadCloser returning interface are defined in a way where if err != nill then data doesnt need a close. As such are we use this commit was needed?